public/js/jibe.js

```diff
@@ -399,6 +408,39 @@ var Jibe = (function (BCSocket, CodeMirror, Replay, showdown, Timestamps, TextFo
       // flag current version
       api.flagVersion ();
     });
+
+    // Table of Contents stuff
+    /* Toggles table of contents panel
+    */
+    $('#toggle-toc').click(function(){
+      if( $('#toc-container').is(":visible")){
+        $('#toc-container').hide('slow',function(){
+          $('#editor-preview-container').removeClass("col-md-9").promise().done(function(){});
```

Comment:
From my experience with promises and callbacks, adding a done() callback to a promise does not mean you're waiting for the promise to finish before continuing. You're registering a callback for when the operation finishes. Normal execution should continue immediately after the call to done(). If you want to perform an action after a promise is done you need to do it inside of the done callback.